### DISCUSSION TITLE
How do I halt a cloodle?

### DISCUSSION THREAD

**Discussion by xrd** [ianthehenry/toodle.studio#9] [Q&A] [answered]
I was hoping that cloodles behave similarly to toodles in that returning nil would halt them (if I understand toodles correctly). But this code seems to throw an error. Is there a way to tell the cloodle to stop after a certain condition has met? In this case I want to stop the cloodle from running once age > 1000.

```
(defn hypotrochoid [f r R t d]
  (+
    (* d (f (* t (/ (- R r) r))))
    (* (- R r) (f t))))

(def d 50)
(def r 30)
(def R 50)

(cloodle 
  (def t (/ age 100))
  (def x (hypotrochoid cos r R t d))
  (def y (hypotrochoid sin r R t (- d)))
  ( do 
      (if (= 0 (% age 100)) (printf "Age %d" age))
      (if (< age 1000 ) 
        [ x y ] 
        nil
      )
    )
)
```

In the console output, I see this:

```
Age 0
Age 0
Age 100
Age 200
Age 300
Age 400
Age 500
Age 600
Age 700
Age 800
Age 900
Age 1000
illegal yield ((29.6027 -29.1101) nil (1 1 1 1) 1)

```

![stop-cloodle](https://user-images.githubusercontent.com/17064/232146491-79493736-6661-453f-9517-be2acbd7a073.png)

**Comment by xrd** [comment, marked as answer]
This seems to work, perhaps just use `die`?

```
(if (< age 1000 ) 
  [ x y ] 
  (die)
)
```

**Reply by ianthehenry** [reply]
Yeah, I do think that it would make sense if a cloodle returned nil that that would automatically kill it -- I added an issue to track #10
